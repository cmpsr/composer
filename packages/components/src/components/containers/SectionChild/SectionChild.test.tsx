import React from 'react';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';

import { SectionChild } from '.';

describe('SectionChild', () => {
  it('should match snapshot', () => {
    const sectionChild = renderer.create(<SectionChild />);
    const tree = sectionChild.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should render class name', () => {
    const { baseElement } = render(<SectionChild className="class-name" />);
    expect(baseElement.innerHTML).toContain('class-name');
  });
  it('should render children', () => {
    const children = <div data-testid="children-test-id">Hello</div>;
    const { getByTestId } = render(<SectionChild>{children}</SectionChild>);
    expect(getByTestId('children-test-id')).toBeInTheDocument();
  });
});
