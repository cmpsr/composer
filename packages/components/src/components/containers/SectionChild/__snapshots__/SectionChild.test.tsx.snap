// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SectionChild should match snapshot 1`] = `
<div
  className="py-8"
/>
`;
